test/unit/view-layout/new-test-layout.cpp

```diff
@@ -0,0 +1,227 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include "gtest/gtest.h"
+
+template<typename T>
+class LayoutTest : public ::testing::Test {};
+
+template<typename T>
+class OffsetLayoutTest : public ::testing::Test {};
+
+using MyTypes = ::testing::Types<RAJA::Index_type,
+				 short,
+				 unsigned short,
+				 int,
+				 unsigned int,
+				 long,
+				 unsigned long,
+				 long int,
+				 unsigned long int,
+				 long long,
+				 unsigned long long>;
+
+
+TYPED_TEST_CASE(LayoutTest, MyTypes);
+TYPED_TEST_CASE(OffsetLayoutTest, MyTypes);
+
+TYPED_TEST(LayoutTest, Constructors)
```

Comment:
Please label all the test cases with "UnitTest" so we can easily distinguish from non-unit tests and filter if needed. So this would be LayoutUnitTest.  Please see how all the other unit tests are labeled in the feature/test-reorg branch.